### ISSUE TITLE
ds-api-build step fails in manual deployment via Deployment manager

### ISSUE THREAD

**Issue by n0531m** [GoogleCloudPlatform/datashare-toolkit#516]
**Describe the bug**
env : 0.7.0 manual deployment via Deployment manager. 

ds-api-build step fails with the following error. 

`{"ResourceType":"gcp-types/cloudbuild-v1:cloudbuild.projects.builds.create","ResourceErrorCode":"504","ResourceErrorMessage":"DEADLINE_EXCEEDED"}
`

Looking at details on Cloud Build the following build has failed. 

Name : `gcr.io/cloud-builders/gcloud`
Arguments : `gcloud alpha run deploy ds-api --cluster=datashare --cluster-location=us-central1-a --namespace=datashare-apis --min-instances=1 --image=gcr.io/XXXXXXXXXX/ds-api:dev --platform=gke --service-account=ds-api-mgr --set-env-vars=UI_BASE_URL=https://datashare.XXXXXXXXX,PROJECT_ID=XXXXXXXXX`
Directory : `ds/datashare-toolkit`

and step 3 shows 
```
ERROR: (gcloud.alpha.run.deploy) [Errno 111] Connection refused
This may be due to network connectivity issues. Please check your network settings, and the status of the service you are trying to reach.
```

On GKE side

Workload `ds-api-00001-zon-deployment` has the following status
`Internal error occurred: failed calling webhook "sidecar-injector.istio.io": Post https://istio-sidecar-injector.gke-system.svc:443/inject?timeout=30s: no endpoints available for service "istio-sidecar-injector"` can be observed.

and `istio-sidecar-injector` has state of `Does not have minimum availability`

adding notes to the GKE node pool did not solve this.  
Any guidance on what to look for is appreciated.
Probally ideal if we can just deploy to Cloud Run without any GKE.

**Comment by swilliams11** [comment]
Based on the other document that you sent to me. I think these errors are related to a permission issue.  What I plan to do is determine if I can add a function in the VM startup script that will check for the appropriate permissions. 

We still need to determine why the permissions that you set on your service account do not allow the VM startup script to apply the Istio policies.  Let's talk again on Monday. Please schedule a call so we can review your setup again.

**Comment by swilliams11** [comment]
Closing ticket since this is resolved and Datashare was installed into your cluster successfully. It turned out to be a permission issue on the custom service account. 

We also ran into an unusual error, where Istio for Cloud Run was not installed on the GKE cluster and that caused the VM startup script to fail, because it could not retrieve the Istio version and package.  Once we deleted the deployment, downloaded 0.7.0 zip file again and reran the deployment, then it setup everything successfully.
